examples/layers/preprocessing/random_shear_bbox_demo.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""random_shear_bboxdemo.py shows how to use the RandomShear preprocessing layer
+   for object detection.
+Operates on the voc dataset.  In this script the images and bboxes
+are loaded, then are passed through the preprocessing layers.
+Finally, they are shown using matplotlib.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from keras_cv.layers import preprocessing
+
+IMG_SIZE = (256, 256)
+BATCH_SIZE = 9
+
+
+def resize(inputs):
+    """mapping function to create batched image and bbox coordinates"""
+    inputs["image"] = tf.image.resize(inputs["image"], IMG_SIZE)[0]
+    height, width, _ = inputs["image"].shape
+    y1, x1, y2, x2 = tf.split(inputs["objects"]["bbox"][0], 4, axis=1)
+    inputs["objects"]["bbox"] = tf.squeeze(
+        tf.stack(
+            [
+                y1 * (IMG_SIZE[0] / height),
+                x1 * (IMG_SIZE[1] / width),
+                y2 * (IMG_SIZE[0] / height),
+                x2 * (IMG_SIZE[1] / width),
+            ],
+            axis=1,
+        ),
+        axis=-1,
+    )
+    return inputs
+
+
+def main():
+    dataset = tfds.load(
+        "voc/2007", split=tfds.Split.TRAIN, batch_size=1, shuffle_files=True
+    )
+    dataset = dataset.map(lambda x: resize(x))
+    dataset = dataset.padded_batch(BATCH_SIZE)
+
+    randomshear = preprocessing.RandomShear(
+        x_factor=(0, 0.2), y_factor=(0, 0.2), fill_mode="constant"
+    )
+    colors = np.array([[0.0, 255.0, 0.0]])
+    for example in dataset.take(1):
+        result = randomshear(
+            {"images": example["image"], "bounding_boxes": example["objects"]["bbox"]}
+        )
+        images, bboxes = result["images"], result["bounding_boxes"]
+        plotted_images = tf.image.draw_bounding_boxes(images, bboxes, colors)
```

Comment:
Having an option to put text is important. https://github.com/keras-team/keras-cv/pull/373#issuecomment-1111313645

Reply:
will add that parameter @innat

Reply:
> will add that parameter @innat

Again, it is not so trivial if we don't want to add `tf.io` as dependency and use its ops version. It is not the only alternative  of course but check the PR content at https://github.com/tensorflow/io/pull/254 as it introduced a c++ dependency over freetype lib. /cc @yongtang

Reply:
Yeah, I did not realize that tf.io was installed separately from tf.  We don't want to use tf.io as a dependency.

Reply:
I don't see too much alternatives if we don't want to extend the TF core API we already have with a de-facto upstreaming PR on what we have in TF IO. 
Any other Keras-cv only alternative will need to handle python or numpy objects e.g. https://github.com/rougier/freetype-py/blob/master/examples/hello-world.py.  
I think this will introduce more or less significant performance impacts.